p11-kit/rpc-server.c

```diff
@@ -251,32 +253,52 @@ proto_read_attribute_buffer (p11_rpc_message *msg,
 	if (!p11_rpc_buffer_get_uint32 (msg->input, &msg->parsed, &n_attrs))
 		return PARSE_ERROR;
 
+	/* Is attrs pointer null or not ? */
+	if (!p11_rpc_buffer_get_byte (msg->input, &msg->parsed, &is_attrs))
+		return PARSE_ERROR;
+
+	if (!is_attrs) {
+		*result = NULL;
+		*n_result = n_attrs;
+		return CKR_OK;
+	}
+
+	if (n_attrs == 0)
+		size_attrs = 1;
```

Comment:
Why this adjustment?

Reply:
Because:
`/* we need at least one (type, value) tuple to be filled */`

I have added this comment, update will come in a minute.

Reply:
For the same reasoning as ulong buffer, I don't understand why we need it. If there is no attribute in the buffer, we don't need to store any element. If the caller is accessing an element even when the buffer length is 0, it is obviously a programming error.

Unless we come up with a scenario (ideally with a test case) that demonstrates the necessity, I would suggest dropping it for now.

Reply:
You are right. I spent some times in re-reading the Oasis spec today: it cannot happen for any functions that use CK_ATTRIBUTE_PTR.

I'll provide an update that remove this part.

Reply:
Please, note that I have aligned the attrs_is_null too. So all logics should be consistent.